crates/ibc/src/core/ics02_client/client_state.rs

```diff
@@ -116,97 +109,27 @@ pub trait ClientState:
         upgraded_consensus_state: Any,
     ) -> Result<UpdatedState, ClientError>;
 
-    /// Verification functions as specified in:
-    /// <https://github.com/cosmos/ibc/tree/master/spec/core/ics-002-client-semantics>
-    ///
-    /// Verify a `proof` that the consensus state of a given client (at height `consensus_height`)
-    /// matches the input `consensus_state`. The parameter `counterparty_height` represent the
-    /// height of the counterparty chain that this proof assumes (i.e., the height at which this
-    /// proof was computed).
-    fn verify_client_consensus_state(
-        &self,
-        proof_height: Height,
-        counterparty_prefix: &CommitmentPrefix,
-        proof: &CommitmentProofBytes,
-        root: &CommitmentRoot,
-        client_cons_state_path: &ClientConsensusStatePath,
-        expected_consensus_state: &dyn ConsensusState,
-    ) -> Result<(), ClientError>;
-
-    /// Verify a `proof` that a connection state matches that of the input `connection_end`.
-    fn verify_connection_state(
-        &self,
-        proof_height: Height,
-        counterparty_prefix: &CommitmentPrefix,
-        proof: &CommitmentProofBytes,
-        root: &CommitmentRoot,
-        counterparty_conn_path: &ConnectionPath,
-        expected_counterparty_connection_end: &ConnectionEnd,
-    ) -> Result<(), ClientError>;
-
-    /// Verify a `proof` that a channel state matches that of the input `channel_end`.
-    fn verify_channel_state(
-        &self,
-        proof_height: Height,
-        counterparty_prefix: &CommitmentPrefix,
-        proof: &CommitmentProofBytes,
-        root: &CommitmentRoot,
-        counterparty_chan_end_path: &ChannelEndPath,
-        expected_counterparty_channel_end: &ChannelEnd,
-    ) -> Result<(), ClientError>;
-
-    /// Verify the client state for this chain that it is stored on the counterparty chain.
-    fn verify_client_full_state(
+    // Verify_membership is a generic proof verification method which verifies a
+    // proof of the existence of a value at a given Path at the specified height.
+    fn verify_membership(
         &self,
         proof_height: Height,
```

Comment:
At first glance it seems odd to me that we are doing a height check as part of `verify_membership`, where the name suggests we're only doing a Merkle tree proof verification.

Reply:
It is fixed because of refactoring height validations in [6258916](https://github.com/cosmos/ibc-rs/pull/531/commits/6258916640f39e7ec417145646df32308c0f40f0)